zookeeper-server/src/test/java/org/apache/zookeeper/server/backup/storage/impl/LocalBackupStorageTest.java

```diff
@@ -0,0 +1,107 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.server.backup.storage.impl;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.zookeeper.common.ConfigException;
+import org.apache.zookeeper.server.backup.BackupConfig;
+import org.apache.zookeeper.server.backup.storage.BackupStorageProvider;
+import org.apache.zookeeper.server.backup.storage.BackupStorageUtil;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runners.MethodSorters;
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class LocalBackupStorageTest {
+  private static BackupConfig backupConfig;
+  private static String testFolderPath;
+  private static String testFileDirPath;
+  private static String backupFolderPath;
+  private static List<String> testFileNames =
+      new ArrayList<>(Arrays.asList("log.testFile_0", "snapshot.testFile_1", "log.testFile_2"));
+  private static BackupStorageProvider backupStorage;
+  private static List<File> testFiles = new ArrayList<>();
+  private List<File> backupFiles = new ArrayList<>();
+
+  @BeforeClass
+  public static void beforeClass() throws IOException, ConfigException {
+    backupConfig = new BackupConfig.Builder().setEnabled(true)
+        .setStatusDir(new File("/localBackupStorageTest/backup/status")).
+            setTmpDir(new File("/localBackupStorageTest/tmp/backup"))
+        .setStorageProviderClassName(LocalBackupStorage.class.getName())
+        .setMountPath("/localBackupStorageTest").setNamespace("test").build().get();
+
+    backupStorage = new LocalBackupStorage(backupConfig);
+    testFolderPath =  Paths.get(Paths.get("").toAbsolutePath().toString(), backupConfig.getMountPath()).toString();
+    testFileDirPath = String.join(File.separator, testFolderPath, "testFiles");
+    backupFolderPath = String.join(File.separator, testFolderPath, "backupStorage");
+
+    for (String testFileName : testFileNames) {
+      String testFilePath = String.join(File.separator, testFileDirPath, testFileName);
+      File testFile = new File(testFilePath);
+      testFiles.add(testFile);
+      BackupStorageUtil.createFile(testFile, false);
+      FileWriter fileWriter = new FileWriter(testFile);
+      for (int i = 0; i < 1000; i++) {
+        fileWriter.write(testFileName);
+        fileWriter.write('\n');
+      }
+      fileWriter.close();
+      Assert.assertTrue(testFile.exists());
+      Assert.assertTrue(
+          Files.readAttributes(Paths.get(testFilePath), BasicFileAttributes.class).size() > 0);
+    }
+  }
+
+  @Test
+  public void test1_GetBackupFileInfo() {
```

Comment:
Do we need to prepend each test method with "testN_"? We could probably remove the prefixes.

Reply:
There are better ways to achieve this. Can you do a bit of research into using either 1) `dependsOnMethods` or 2) TestNG priority?

Reply:
Looks like ZooKeeper uses junit instead of TestNG, so the two options you mentioned are not available. I'll keep the current implementation.